src/lib/server/query_handler.cpp

```diff
@@ -96,7 +96,7 @@ void QueryHandler::handle_transaction_statement_message(ExecutionInformation& ex
   auto& statements = sql_statement->getStatements();
 
   if (statements.front()->isType(hsql::StatementType::kStmtTransaction)) {
-    auto* transaction_statement = reinterpret_cast<hsql::TransactionStatement*>(statements.front());
+    auto* transaction_statement = dynamic_cast<hsql::TransactionStatement*>(statements.front());
```

Comment:
Since you are not testing this for `nullptr`, `(const) auto& ... = dynamic_cast<...&>(*statements.front())` would be the canonical way.

Reply:
Done.